packages/panzer/mini-em/example/BlockPrec/main.cpp

```diff
@@ -0,0 +1,691 @@
+      std::string strategy = "Constant";
+      Teuchos::ParameterList p;
+      p.set("Value X",0.0);
+      p.set("Value Y",0.0);
+      p.set("Value Z",0.0);
+      panzer::BC bc(bc_id, bctype, sideset_id, element_block_id, dof_name, 
+                    strategy, p);
+      bcs.push_back(bc);
+      bc_id++;
+    }
+ 
+  return bcs;
+}
+
+//! Create BCs for auxiliary operators
+std::vector<panzer::BC> auxiliaryBoundaries()
+{
+  std::vector<panzer::BC> bcs;
+
+  std::string sidesets[6] = {"left","right","top","bottom","back","front"};
+  std::string eq_sets[2]  = {"Mass Matrix AUXILIARY_NODE","Weak Gradient"};
+  std::string dofs[2]     = {"AUXILIARY_NODE","AUXILIARY_EDGE"};
+
+  std::size_t bc_id = 0;
+  for (int s = 0; s < 6; s++)
+    for (int d = 0; d < 2; d++)
+    {
+      panzer::BCType bctype = panzer::BCT_Dirichlet;
+      std::string sideset_id = sidesets[s];
+      std::string element_block_id = "eblock-0_0_0";
+      std::string dof_name = eq_sets[d];
+      std::string strategy = "AuxConstant";
+      Teuchos::ParameterList p;
+      p.set("Field Name",dofs[d]);
+      p.set("Value",0.0);
+      panzer::BC bc(bc_id, bctype, sideset_id, element_block_id, dof_name, 
+                    strategy, p);
+      bcs.push_back(bc);
+      bc_id++;
+    }
+ 
+  return bcs;
+}
+
+//! Create parameter list defining nodal mass matrix and node-edge weak gradient
+Teuchos::RCP<Teuchos::ParameterList> auxOpsParameterList(const int basis_order)
+{
+  Teuchos::RCP<Teuchos::ParameterList> pl = Teuchos::rcp(new Teuchos::ParameterList("Aux Physics Block"));
+  const int integration_order = 2*basis_order;
+
+  {
+    Teuchos::ParameterList& p = pl->sublist("Auxiliary Mass Physics");
+    p.set("Type","Auxiliary Mass Matrix");
+    p.set("DOF Name","AUXILIARY_NODE");
+    p.set("Basis Type","HGrad");
+    p.set("Basis Order",basis_order);
+    p.set("Integration Order",integration_order);
+  }
+
+  {
+    Teuchos::ParameterList& p = pl->sublist("Auxiliary Gradient Physics");
+    p.set("Type","Auxiliary Weak Gradient");
+    p.set("Vector Name","AUXILIARY_EDGE");
+    p.set("Scalar Name","AUXILIARY_NODE");
+    p.set("Basis Order",basis_order);
+    p.set("Integration Order",integration_order);
+  }
+
+  return pl;
+}
+
+//! Create parameter list defining linear solver
+Teuchos::RCP<Teuchos::ParameterList> maxwellSolverParameterList(const bool use_ilu, const bool use_refmaxwell, const bool print_diagnostics)
+{
+  Teuchos::RCP<Teuchos::ParameterList> pl = Teuchos::rcp(new Teuchos::ParameterList("Linear Solver"));
+  pl->set("Linear Solver Type","Belos");
+  {
+    Teuchos::ParameterList& ls_types = pl->sublist("Linear Solver Types");
+    {
+      Teuchos::ParameterList& belos = ls_types.sublist("Belos");
+      belos.set("Solver Type","Block GMRES");
+      {
+        Teuchos::ParameterList& solver_types = belos.sublist("Solver Types");
+        {
+          Teuchos::ParameterList& gmres = solver_types.sublist("Block GMRES");
```

Comment:
Do you actually use the "Block" feature of "Block GMRES"?  If not, it suffices to ask Belos' factory for "GMRES".